src/compat/libc/string/memccpy.c

```diff
@@ -0,0 +1,67 @@
+/**
+ *  @file
+ *  @brief Implementation of #memccpy() function.
+ *
+ *  @date 18/3/20
+ *  @author Giannis Zapantis (github name: tech-gian)
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+void* memccpy(void* dest, const void* src, int c, size_t n) {
+
+/* If overlap happens the results
+ * will be undefined.
+ */
+
+/* Getting dest and srt into byte
+ * format.
+ */
+    char* temp_dest;
+    char* temp_src;
+    size_t pos;
+    
+    temp_dest = (char*) dest;
+    temp_src = (char*) src;
+    pos = n;
+
+    char* p_src[n];
+
+    size_t i;
+
+    for (i = 0 ; i < n ; i++) {
+        if (temp_src[i] == c) {
+            pos = i;
+            break;
+        }
+    }
+
+/* getting the pointers from src
+ * into an array of pointers.
+ */
+    for (i = 0 ; i < n ; i++) {
+        p_src[i] = &(temp_src[i]);
+    }
+
+/* copying the value from src
+ * to dest until c found.
+ */
+    for (i = 0 ; i < n ; i++) {
+        if (i < pos) {
+            temp_dest[i] = *p_src[i];
+        }
+        else {
+            return dest+i;
+        }
+    }

```

Comment:
Why do you need these 3 loops? You can iterate char by char and copy `n` chars from `src` to `dest` only if the current char is not `c`. And if the current char is `c` you just exit from the loop.

Reply:
I don't need 3 loops, you are right. I replaced it with one loop doing exactly what you are telling me.

Reply:
Ok, now it looks better.